packages/cozy-harvest-lib/test/components/AccountLogoutButton.spec.js

```diff
@@ -0,0 +1,83 @@
+import React from 'react'
+import { shallow } from 'enzyme'
+
+import { AccountLogoutButton } from 'components/AccountLogoutButton'
+
+const fixtures = {
+  account: {
+    _id: '61c683295560485db0f34b859197c581',
+    account_type: 'konnectest',
+    auth: {
+      username: 'foo',
+      passphrase: 'bar'
+    },
+    identifier: 'username'
+  }
+}
+
+const tMock = () => 'Mock button label'
+
+const props = {
+  t: tMock,
+  deleteAccount: jest.fn()
+}
+
+describe('AccountLogoutButton', () => {
+  beforeEach(() => {
+    jest.resetAllMocks()
+  })
+
+  it('should render without listeners', () => {
+    const component = shallow(
+      <AccountLogoutButton account={fixtures.account} {...props} />
+    ).getElement()
+    expect(component).toMatchSnapshot()
+  })
+
+  it('should run deleteAccount on click', () => {
+    const component = shallow(
+      <AccountLogoutButton account={fixtures.account} {...props} />
+    )
+    component.simulate('click')
+    expect(props.deleteAccount).toHaveBeenCalledWith(fixtures.account)
+    // the rest is asserted in the onSuccess
+  })
+
+  it('should change state on click and handle onSuccess', done => {
```

Comment:
I don't think we should test state here. It's internal implementation and we should not care about that.

Instead, I suggest to test the that the button is rendered with a disabled and busy props.